redash/query_runner/corporate_memory.py

```diff
@@ -0,0 +1,257 @@
+        sparql_results = json_loads(results)
+        # transform all bindings to redash rows
+        rows = []
+        for sparql_row in sparql_results["results"]["bindings"]:
+            row = {}
+            for var in sparql_results["head"]["vars"]:
+                try:
+                    row[var] = sparql_row[var]["value"]
+                except KeyError:
+                    # not bound SPARQL variables are set as empty strings
+                    row[var] = ""
+            rows.append(row)
+        # transform all vars to redash columns
+        columns = []
+        for var in sparql_results["head"]["vars"]:
+            columns.append({"name": var, "friendly_name": var, "type": "string"})
+        # Not sure why we do not use the json package here but all other
+        # query runner do it the same way :-)
+        return json_dumps({"columns": columns, "rows": rows})
+
+    @classmethod
+    def name(cls):
+        return "eccenca Corporate Memory (SPARQL)"
+
+    @classmethod
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def type(cls):
+        return "corporate_memory"
+
+    def run_query(self, query, user):
+        """send a sparql query to corporate memory"""
+        query_text = query
+        logger.info("about to execute query (user='{}'): {}".format(user, query_text))
+        query = SparqlQuery(query_text)
+        query_type = query.get_query_type()
+        # type of None means, there is an error in the query
+        # so execution is at least tried on endpoint
+        if query_type not in ["SELECT", None]:
+            raise ValueError(
+                "Queries of type {} can not be processed by redash.".format(query_type)
+            )
+
+        self._setup_environment()
+        try:
+            data = self._transform_sparql_results(query.get_results())
+        except Exception as error:
+            logger.info("Error: {}".format(error))
+            try:
+                # try to load Problem Details for HTTP API JSON
+                details = json.loads(error.response.text)
+                error = ""
+                if "title" in details:
+                    error += details["title"] + ": "
+                if "detail" in details:
+                    error += details["detail"]
+                    return None, error
+            except Exception:
+                pass
+
+            return None, error
+
+        error = None
+        return data, error
+
+    @classmethod
+    def configuration_schema(cls):
+        """provide the configuration of the data source as json schema"""
+        return {
+            "type": "object",
+            "properties": {
+                "CMEM_BASE_URI": {"type": "string", "title": "CMEM_BASE_URL"},
+                "OAUTH_GRANT_TYPE": {
+                    "type": "string",
+                    "title": "OAUTH_GRANT_TYPE (can be: password or client_credentials)",
+                    "default": "client_credentials",
```

Comment:
We support dropdowns, so maybe better to use those to make sure people don't misspell?

Reply:
sure, exactly what is needed here. Will do

Reply:
@arikfr There quite some extensions to the schema description system compared to v8 (at lease I was not aware of it), so I not only introduced a dropdown but also a checkbox and the additional settings sections.

Hope this can be merged now.

btw: please also be aware of https://github.com/getredash/redash/pull/5415/commits/56349d4788d20d9026ca58519023c0f3d3fc4066